interfaces/builtin/login_session_control.go

```diff
@@ -0,0 +1,71 @@
+// -*- Mode: Go; indent-tabs-mode: t -*-
+
+/*
+ * Copyright (C) 2019 Canonical Ltd
+ *
+ * This program is free software: you can redistribute it and/or modify
+ * it under the terms of the GNU General Public License version 3 as
+ * published by the Free Software Foundation.
+ *
+ * This program is distributed in the hope that it will be useful,
+ * but WITHOUT ANY WARRANTY; without even the implied warranty of
+ * MERCHANTABILITY or FITNESS FOR A PARTICULAR PURPOSE.  See the
+ * GNU General Public License for more details.
+ *
+ * You should have received a copy of the GNU General Public License
+ * along with this program.  If not, see <http://www.gnu.org/licenses/>.
+ *
+ */
+
+package builtin
+
+const loginSessionControlSummary = `allows setup of login session & seat`
+
+const loginSessionControlBaseDeclarationSlots = `
+  login-session-control:
+    allow-installation:
+      slot-snap-type:
+        - core
+    deny-auto-connection: true
+    deny-connection:
+      on-classic: false
```

Comment:
What is the need to allow connection on classic systems? Or am I reading this wrong?

Reply:
Probably I read it wrong while experimenting (this is all unfamiliar syntax to me).

It works without this clause, so I'm just removing it.

Reply:
The question is: what should the default policy be? Should it allow or deny connecting to the interface?

Reply:
I want to enable creating a confined session that can be selected and started from the greeter. That currently works with an unconfined snap as it can negotiate with logind, from a confined snap the negotiation requires something like this interface.

Reply:
Let me try rephrasing:

On classic systems without this interface, a confined snap does not have the access needed to provide a graphical non-root shell.

That means the interface is useful on classic.

An example would be the "egmde-confined-desktop" snap, which currently runs as a daemon (i.e. root) as a way to gain the access it needs. That has the unfortunate side-effect "taking over" whenever the snap starts.

With this interface in place it will be possible to run an updated  "egmde-confined-desktop" snap as a strictly confined user session selected from the greeter.